modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/LiferayObjectWrapper.java

```diff
@@ -234,6 +211,64 @@ else if (object instanceof ResourceBundle) {
 		return modelFactory.create(object, this);
 	}
 
+	private void _checkClassIsRestricted(Class<?> clazz)
+		throws TemplateModelException {
+
+		String className = clazz.getName();
+
+		ClassRestrictionInformation classRestrictionInformation =
+			_classRestrictionInformations.computeIfAbsent(
+				className,
+				a -> {
+					if (_allowedClassNames.contains(className)) {
```

Comment:
@csierra This checking does make sense to me. If the classes is in allowed list, you should not block it.

Reply:
hey @tinatian you are absolutely right... it looks I turned the test around when I cached that
fixing...
thx!